packages/enterprise/src/power.ts

```diff
@@ -234,27 +244,52 @@ export function calculateMidExperimentPowerSingle(
     );
   } else {
     /*freq power*/
-    let halfwidth: number;
     if (params.sequential) {
-      const s2 = sigmahat2Delta * firstPeriodPairwiseSampleSize;
-      const nTotal =
-        firstPeriodPairwiseSampleSize * (scalingFactorLowPowerWarning + 1);
-      halfwidth = sequentialIntervalHalfwidth(
-        s2,
-        nTotal,
-        params.sequentialTuningParameter,
-        params.alpha / numTests
-      );
+      if (response.deltaPosterior === undefined) {
```

Comment:
Is it worth it to create two different interfaces upstream? One for non-sequential power and one for sequential power? Then we don't need to have all these undefined checks as we will rely on TS to ensure we pass the needed fields for the sequential version.

Or is this risky because the `response` is coming from the stats engine and we can't be sure all of the data exists?